File: kinotic-js/workspace/packages/os-api/src/api/model/github/AvailableRepo.ts
/**
 * One row in the repo dropdown shown when linking a project. Carries just enough
 * info for the UI; a successful link round-trips through
 * {@link IProjectGitHubRepoService} which re-validates against GitHub before
 * persisting.
 */
export class AvailableRepo {
    public repoId: number = 0
    public fullName: string = ''
    public defaultBranch: string = ''
    public privateRepo: boolean = false
}

File: kinotic-github/src/main/java/org/kinotic/github/api/model/AvailableRepo.java
package org.kinotic.github.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * One row in the repo dropdown shown when linking a project. Carries just enough info
 * for the UI; a successful link round-trips through {@code ProjectGitHubRepoService}
 * which re-validates against GitHub before persisting.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class AvailableRepo {
    private Long repoId;
    private String fullName;
    private String defaultBranch;
    private boolean privateRepo;
}
